test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1

```diff
@@ -0,0 +1,93 @@
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+    }
+
+    It "Get-Module fails if not using -ListAvailable with '<parameter>'" -TestCases @(
+        @{parameter="PSSession" ; value=$pssession},
+        @{parameter="CIMSession"; value=$cimsession}
+    ) {
+        param($parameter, $value)
+        $parameters = @{$parameter=$value}
+        { Get-Module @parameters } | ShouldBeErrorId "RemoteDiscoveryWorksOnlyInListAvailableMode,Microsoft.PowerShell.Commands.GetModuleCommand"
+    }
+
+    It "Get-Module succeeds using -ListAvailable with '<parameter>'" -TestCases @(
+        @{parameter="PSSession" ; value=$pssession},
+        @{parameter="PSSession" ; value=$pssession ; name="Pester"},
+        @{parameter="CIMSession"; value=$cimsession},
+        @{parameter="CIMSession"; value=$cimsession; name="pESTER"}
+    ) {
+        param($parameter, $value, $name)
+        $parameters = @{$parameter=$value; ListAvailable=$true; Refresh=$true}
+        if ($name) {
+            $parameters += @{name=$name}
+        }
+        $modules = Get-Module @parameters
+        $modules | Should Not BeNullOrEmpty
+        $modules[0] | Should BeOfType "System.Management.Automation.PSModuleInfo"
+    }
+
+    It "Get-Module can be called as an API with '<parameter>' = '<value>'" -TestCases @(
+        @{parameter = "Name"              ; value = "foo"},
+        @{parameter = "FullyQualifiedName"; value = @{ModuleName="foo";ModuleVersion="1.2.3"}},
+        @{parameter = "All"               ; value = $true},
+        @{parameter = "All"               ; value = $false},
+        @{parameter = "ListAvailable"     ; value = $true},
+        @{parameter = "ListAvailable"     ; value = $false},
+        @{parameter = "PSEdition"         ; value = "foo"},
+        @{parameter = "Refresh"           ; value = $true},
+        @{parameter = "Refresh"           ; value = $false},
+        @{parameter = "PSSession"         ; value = $pssession},
+        @{parameter = "CimSession"        ; value = $cimsession},
+        @{parameter = "CimResourceUri"    ; value = "http://foo/"},
+        @{parameter = "CimNamespace"      ; value = "foo"},
+        @{parameter = "PSEdition"         ; value = "Core"},
+        @{parameter = "PSEdition"         ; value = "Desktop"}
+        ) {
+        param($parameter, $value)
+
+        $getModuleCommand = [Microsoft.PowerShell.Commands.GetModuleCommand]::new()
+        $getModuleCommand.$parameter = $value
+        if ($parameter -eq "FullyQualifiedName") {
+            $getModuleCommand.FullyQualifiedName | Should BeOfType "Microsoft.PowerShell.Commands.ModuleSpecification"
+            $getModuleCommand.FullyQualifiedName.Name | Should Be "foo"
+            $getModuleCommand.FullyQualifiedName.Version | Should Be "1.2.3"
+        } else {
+            $getModuleCommand.$parameter | Should Be $value
+        }
+    }
+
+    It "ArugmentCompleter for PSEdition should work for '<cmd>'" -TestCases @(
```

Comment:
I see currently we use `TabExpansion2 ` only in two files. Should we put completion tests in one place or put tab completion tests in cmdlet test files?
Make sense make one meta test to cover all (cmdlets) tab completions? It seems something like we do for aliases.

Reply:
I'll move this to TabCompletion.Tests.ps1